fix: disable basic auth for HTTP API (#9586)
emqx/emqx: changes/v5.0.15/feat-9586.en.md

> The `basic` authentication which based on the `username:password` has no longer supported anymore, now the API must call with the API key.

Reviewer: ```suggestion
Basic HTTP Authentication (`username:password`) is not supported by HTTP API anymore, all API calls must use `Bearer` scheme and an API key.
```

> The `basic` authentication which based on the `username:password` has no longer supported anymore, now the API must call with the API key.

Reviewer 2: ```suggestion
Basic auth is no longer allowed for API calls, must use API key instead.
```
